### PR TITLE
fix delete e2e workspace

### PR DESCRIPTION
**Pull request by tamirkamara**
## What is being addressed

Cleanup script always skips deleting old e2e workspaces.

## How is this addressed

- Adjust the running workflow condition to not check the cleanup workflow itself before deciding to delete workspaces.

### CODE REVIEW

**Comment by tamirkamara** [comment]
/test-force-approve
This change isn't checked via the tests.

File: devops/scripts/clean_ci_validation_envs.sh [hunk 1]
```diff
@@ -65,14 +65,16 @@ while read -r rg_name rg_ref_name; do
   fi
 done
 
-# check if any workflows run on the main branch
+# check if any workflows run on the main branch (except the cleanup one)
 # to prevent us deleting a workspace for which an E2E (on main) is currently running
-if [ $(gh api "https://api.github.com/repos/microsoft/AzureTRE/actions/runs?branch=main&status=in_progress" --jq ".total_count") == 0 ]
+if [[ -z $(gh api "https://api.github.com/repos/microsoft/AzureTRE/actions/runs?branch=main&status=in_progress" | jq '.workflow_runs | select(.[].name != "Clean Validation Environments")') ]]
```
Comment (eladiw) [inline devops/scripts/clean_ci_validation_envs.sh:70]: is there a possibility to use id of some kind instead of name? so it won't break if someone rename

Comment (tamirkamara) [reply devops/scripts/clean_ci_validation_envs.sh]: I used the name from the github context which is even better...

**Review by eladiw — APPROVED** [review]

**Pull request merged**